modules/compiler/source/base/source/printf_replacement_pass.cpp

```diff
@@ -335,8 +335,8 @@ std::optional<std::string> getPointerToStringAsString(Value *op) {
       if (const_string->getOpcode() == Instruction::GetElementPtr) {
         var = dyn_cast<GlobalVariable>(const_string->getOperand(0));
       } else if (const_string->getOpcode() == Instruction::IntToPtr) {
-        // SPIR generates a PtrToInt expression inside an IntToPtr expression,
-        // so therefore we need to unwrap it twice.
+        // Sometimes we see a PtrToInt expression inside an IntToPtr
```

Comment:
This said SPIR before - is this still an issue?

Reply:
If it's not incorrect IR I was thinking we should keep the code to handle it. I think the old comment was overly specific to where we've seen the IR coming from.